### DISCUSSION TITLE
Is papi ready yet?

### DISCUSSION THREAD

**Discussion by xlc** [polkadot-api/polkadot-api#337] [General]
I would like to write a quick scripts to connect to relaychain and parachains to subscribe and query data. I also want to give this a try. Is papi ready yet? If not, any estimation timeline? I can wait for like a month, but if longer, than I will just use pjs for now.

Maybe I should wait for #317?

**Comment by josepot** [comment]
> I would like to write a quick scripts to connect to relaychain and parachains to subscribe and query data. I also want to give this a try. Is papi ready yet?

It is definitely ready for these kinds of things, and -in fact- there are some "early adopters" already using it for these kinds of things (Eg [1](https://github.com/paritytech/polkadot-testnet-faucet/blob/main/src/faucet.e2e.ts) and [2](https://github.com/paritytech/get-fellows-action/blob/main/src/fellows.ts) ).

> Is papi ready yet? If not, any estimation timeline? I can wait for like a month, but if longer, than I will just use pjs for now.
> 
> Maybe I should wait for #317?

Our goal is still to release the first beta version (with a "decent" set of docs) by the end of Q1 (or beginning of Q2). Meaning that we hope to have released the first beta version within the next 2-3 weeks.

The thing is that we have been iterating quite a lot on the top-level API (based on the feedback that we have been receiving), and there wasn't much of a point on documenting an API that we knew was not final. That being said, now we have reached a point where we have a very clear idea on which parts are going to remain stable, and which parts are still subject to changes prior to releasing the first stable version.

So, yeah, our goal is still what we stated in the OpenGov proposal (and in a Polkadot forum post): release the first beta (with docs) within the next 2-3 weeks, and then release the first stable version 3 months later.

**Comment by carlosala** [comment]
Closing, PAPI is clearly ready now 🎉 🚀
